library/n2_protocol.hpp

```diff
@@ -8,14 +8,35 @@
 
 namespace ioremap { namespace elliptics { namespace n2 {
 
-struct data_in_file {
-	int fd = -1;
-	off_t local_offset = 0;
-	size_t fsize = 0;
-	int on_exit = 0;
+class data_in_file {
+public:
+	struct info_t {
+		int fd = -1;
+		off_t local_offset = 0;
+		size_t fsize = 0;
+	};
+
+	data_in_file() = default;
+	data_in_file(data_in_file &&) = default;
+	data_in_file &operator = (data_in_file &&) = default;
+
+	~data_in_file();
+
+	const info_t &info() const;
+
+	void detach();
+	data_pointer read_and_detach();
+
+private:
+	data_in_file(const data_in_file &) = delete;
```

Comment:
Could you use `ELLIPTICS_DISABLE_COPY` and add new macros for enabling move?

Reply:
Why you want to forbid copying of this object?

Reply:
on_exit actions must be done once. But I had mistake: default move constructor doesn't solve the problem, I need to write custom constructor.